### PR TITLE
Add /tensor_data endpoint; add tensor overview table

### PR DESCRIPTION
**Pull request by caisq**
The tensor overview table provides summary of debug tensors received
during the Session, including the tensor name, debug op, dtype and
shape. It shows the full values only for tensors with 4 or fewwer
elements. For larger tensors, click the expand icon at the rightmost end
of the row to view a slices of the tensor.

The sliced tensor values are served from the /tensor_data endpoint.
tdp1
tdp2

### CODE REVIEW

**Comment by caisq** [comment]
Revised the PR a little. 
* Separated the tf-tensor-data-view widget into a separate file: tf-tensor-data-view.html
* The table data table no longer creates a new row each time a new tensor data instance arrives. Instead, it displays the unique tensors as rows of the table and highlights the row that corresponds to the last-received tensor data.

See new screenshot:
tdp3

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-tensor-data-view.html [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+      const tensorDataRow = document.createElement('tr');
+      const tensorNameCell = document.createElement('td');
+      tensorNameCell.textContent = tensorName;
+      const countCell = document.createElement('td');
+      countCell.textContent = count;
+      if (isActive) {
+        countCell.style.fontWeight = 'bold';
+      }
+      // Display debug op only if it is not the default (DebugIdentity).
+      // const debugOpCell = document.createElement('td');
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-tensor-data-view.html]: Maybe elaborate on these lines that are commented out.

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-tensor-data-view.html [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      }
+      :host .tensor-data-table {
+        align-content: left;
+        align-items: left;
+        text-align: left;
+        border-style: solid 1px black;
+        width: 100%;
+      }
+      .active-tensor {
+        background-color: lightyellow;
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-tensor-data-view.html]: Maybe use #FFFFE0.

File: tensorboard/plugins/debugger/debugger_plugin.py [hunk 3]
```diff
@@ -540,6 +542,20 @@ def _serve_comm(self, request):
...
+    time_indices = int(request.args.get('time_indices'))
+    # TODO(cais): Handle the case where time_indices is not a single int, i.e.,
+    # a list of int.
+    slicing = request.args.get('slicing')
+    sliced_tensor_data = self._debugger_data_server.query_tensor_store(
+        watch_key, time_indices=time_indices, slicing=slicing)
+    response = {
+        "tensor_data": sliced_tensor_data
+    }
+    return http_util.Respond(request, response, "application/json")
```
[the remark this answers is not in the document]
Comment (caisq) [reply tensorboard/plugins/debugger/debugger_plugin.py]: Done. Fixed throughout this file.

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-tensor-data-view.html (hunk 2 above)
Comment (caisq) [reply tensorboard/plugins/debugger/tf_debugger_dashboard/tf-tensor-data-view.html]: Done.

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-tensor-data-view.html (hunk 1 above)
Comment (caisq) [reply tensorboard/plugins/debugger/tf_debugger_dashboard/tf-tensor-data-view.html]: Done.

**Pull request merged**